app/Jobs/EDocument/CreateEDocument.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2023. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+namespace App\Jobs\EDocument;
+
+use App\Models\Credit;
+use App\Models\Invoice;
+use App\Models\PurchaseOrder;
+use App\Models\Quote;
+use App\Services\EInvoicing\Standards\FacturaEInvoice;
+use App\Services\EInvoicing\Standards\OrderXDocument;
+use App\Services\EInvoicing\Standards\ZugferdEDokument;
+use App\Utils\Ninja;
+use horstoeko\zugferd\ZugferdDocumentBuilder;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\App;
+use function App\Jobs\Invoice\app;
+
+class CreateEDocument implements ShouldQueue
+{
+    use Dispatchable;
+    use InteractsWithQueue;
+    use Queueable;
+    use SerializesModels;
+
+    public $deleteWhenMissingModels = true;
+
+    public function __construct(private object $document, private bool $returnObject = false)
+    {
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return string|ZugferdDocumentBuilder
+     */
+    public function handle(): string|ZugferdDocumentBuilder
+    {
+        /* Forget the singleton*/
+        App::forgetInstance('translator');
+
+        /* Init a new copy of the translator*/
+        $t = app('translator');
+        /* Set the locale*/
+        App::setLocale($this->document->client->locale());
+
+        /* Set customized translations _NOW_ */
+        $t->replace(Ninja::transformTranslations($this->document->client->getMergedSettings()));
```

Comment:
I think the logic breaks here when assuming ->client->

PurchaseOrders are ->vendor->

So if we are using a single entry point for this logic, we'll need to determine if PurchaseOrder and use a prop like $settings_entity

ir

```
$settings_entity = ($this->document Instanceof PurchaseOrder) ? $this->document->vendor : $this->document->client;
```

The you can reference $settings_entity->getSetting() or $settings->entity->getMergedSettings()